### ISSUE TITLE
IntelliSense-based Go to Declaration failed in this case

### ISSUE THREAD

**Issue by lh123** [microsoft/vscode-cpptools#2761]
_Originally posted by @lh123 in https://github.com/Microsoft/vscode-cpptools/issues/2743#issuecomment-434932111_

## 1. Step to reproduce:
> 1. Add `#include <list>`.
> 2. A function named `insert` only defined.

## 2. Code:
```c++
#include <list>//Need include list to reproduce

void insert(int a) {//Need function name is 'insert'

}

int main() {
    insert(0);//Go to Declaration At here.
}
```

```c++
#include <list>

void otherName(int a) {

}

int main() {
    otherName(0); //Go to Declaration is correct.
}

```
## 3. Screenshort:
![image](https://user-images.githubusercontent.com/10697249/47834015-9c509000-ddd8-11e8-905b-ab5aff1c2d7a.png)

**Comment by lh123** [comment]
@sean-mcmanus 
Thank you for your reply.

1. Yes, I am use MinGW-W64 8.2.0 in MSYS2.
2. Before I do Go to **Declaration**, I am sure that the red flame has disappeared.
3. Go to **Definition** works fine.

**Comment by sean-mcmanus** [comment]
The usage of "__catch()" in stl_list.h caused our parser to parse the file incorrectly (i.e. parsing the insert call as a declaration). Adding "#define __catch()" to the cpp.hint file in our extension install location should "workaround" the issue after also doing a Reset IntelliSense Database command. We should have a fix in our Insider release later this week.

![image](https://user-images.githubusercontent.com/19859882/48041196-f763e900-e130-11e8-8104-c9d3f12a118d.png)

**Comment by sean-mcmanus** [comment]
Should be fixed with 0.21.0-insiders (auto installed if `"C_Cpp.updateChannel": "Insiders"` is set, or manually installed from the [offline vsix](https://github.com/Microsoft/vscode-cpptools/releases/tag/0.21.0-insiders)). Let us know if you find any problems with it.
